compute/disks/disks_test.go

```diff
@@ -0,0 +1,270 @@
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package snippets
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"math/rand"
+	"strings"
+	"testing"
+	"time"
+
+	compute "cloud.google.com/go/compute/apiv1"
+	"github.com/GoogleCloudPlatform/golang-samples/internal/testutil"
+	computepb "google.golang.org/genproto/googleapis/cloud/compute/v1"
+	"google.golang.org/protobuf/proto"
+)
+
+func createDisk(ctx context.Context, projectId, zone, diskName, sourceImage string) error {
+	disksClient, err := compute.NewDisksRESTClient(ctx)
+	if err != nil {
+		return err
+	}
+	defer disksClient.Close()
+	req := &computepb.InsertDiskRequest{
+		Project: projectId,
+		Zone:    zone,
+		DiskResource: &computepb.Disk{
+			Name:        proto.String(diskName),
+			SourceImage: proto.String(sourceImage),
+		},
+	}
+
+	op, err := disksClient.Insert(ctx, req)
+	if err != nil {
+		return err
+	}
+
+	return op.Wait(ctx)
+}
+
+func getInstance(
+	ctx context.Context,
+	projectID, zone, instanceName string,
+) (*computepb.Instance, error) {
+	instancesClient, err := compute.NewInstancesRESTClient(ctx)
+	if err != nil {
+		return nil, err
+	}
+	defer instancesClient.Close()
+	reqInstance := &computepb.GetInstanceRequest{
+		Project:  projectID,
+		Zone:     zone,
+		Instance: instanceName,
+	}
+
+	return instancesClient.Get(ctx, reqInstance)
+}
+
+func createDiskSnapshot(ctx context.Context, projectId, zone, diskName, snapshotName string) error {
+	disksClient, err := compute.NewDisksRESTClient(ctx)
+	if err != nil {
+		return err
+	}
+	defer disksClient.Close()
+	req := &computepb.CreateSnapshotDiskRequest{
+		Project: projectId,
+		Zone:    zone,
+		Disk:    diskName,
+		SnapshotResource: &computepb.Snapshot{
+			Name: proto.String(snapshotName),
+		},
+	}
+
+	op, err := disksClient.CreateSnapshot(ctx, req)
+	if err != nil {
+		return err
+	}
+
+	return op.Wait(ctx)
+}
+
+func deleteDiskSnapshot(ctx context.Context, projectId, snapshotName string) error {
+	snapshotsClient, err := compute.NewSnapshotsRESTClient(ctx)
+	if err != nil {
+		return err
+	}
+	defer snapshotsClient.Close()
+	req := &computepb.DeleteSnapshotRequest{
+		Project:  projectId,
+		Snapshot: snapshotName,
+	}
+
+	op, err := snapshotsClient.Delete(ctx, req)
+	if err != nil {
+		return err
+	}
+
+	return op.Wait(ctx)
+}
+
+func deleteInstance(ctx context.Context, projectId, zone, instanceName string) error {
+	instancesClient, err := compute.NewInstancesRESTClient(ctx)
+	if err != nil {
+		return err
+	}
+	defer instancesClient.Close()
+	req := &computepb.DeleteInstanceRequest{
+		Project:  projectId,
+		Zone:     zone,
+		Instance: instanceName,
+	}
+
+	op, err := instancesClient.Delete(ctx, req)
+	if err != nil {
+		return err
+	}
+
+	return op.Wait(ctx)
+}
+
+func TestComputeDisksSnippets(t *testing.T) {
```

Comment:
Can this be split up into individual tests with a TestMain for setup? Or use t.Run to help break this up. Hard to follow what is being tested as is.